ui/index.d.ts

```diff
@@ -580,8 +580,11 @@ export interface IErrorHandlingContext {
 }
 
 export interface ITelemetryReporter {
-    sendTelemetryEvent(eventName: string, properties?: { [key: string]: string | undefined }, measurements?: { [key: string]: number | undefined }): void;
-
+    /**
+     * Use this method for sending error telemetry as traditional events to App Insights.
+     * This method will automatically drop error properties in certain environments for first party extensions.
```

Comment:
Somewhat confusing...  Do these come straight from vscode telemetry?

> This method will automatically drop error properties in certain environments for first party extensions.

Doesn't seem automatic since last arg is required to drop error properties, otherwise all are dropped?

>   If no array is passed, we will drop all properties but still send the event.

I thought the behavior was to drop the entire event?

Reply:
BTW, thanks for handling this!

Reply:
Yeah I copied this straight from here: https://github.com/microsoft/vscode-extension-telemetry#sending-errors-as-events

> Doesn't seem automatic since last arg is required to drop error properties, otherwise all are dropped?

It _is_ automatic. If you forget to pass an empty array for error properties, you don't get any properties (in those certain environments). Source code [here](https://github.com/microsoft/vscode-extension-telemetry/blob/master/src/telemetryReporter.ts#L250)

> I thought the behavior was to drop the entire event?

Nope just the properties. See above link to source code